component/custom/custom_loader.go

```diff
@@ -1,40 +1,67 @@
 package custom
 
 import (
+	"fmt"
 	"os"
+	"path/filepath"
 	"plugin"
 	"strings"
+
+	"github.com/Azure/azure-storage-fuse/v2/common/log"
+	"github.com/Azure/azure-storage-fuse/v2/external"
+	"github.com/Azure/azure-storage-fuse/v2/internal"
 )
 
 func init() {
-	pluginPath := os.Getenv("PLUGIN_PATH")
-	if pluginPath == "" {
+	// Environment variable for directory containing the plugin .so files.
+	pluginDirPath := os.Getenv("BLOBFUSE_PLUGIN_DIRECTORY")
+	if pluginDirPath == "" {
 		return
 	}
 
-	// environment variable can contain multiple paths separated by colon.
-	paths := strings.Split(pluginPath, ":")
+	// Environment variable which expects file names as a colon-separated list of `.so` files located in
+	// the specified plugin directory. For example: BLOBFUSE_PLUGIN_NAMES=plugin1.so:plugin2.so:plugin3.so
+	pluginFilesPath := os.Getenv("BLOBFUSE_PLUGIN_NAMES")
 
-	for _, file := range paths {
-		println("custom_loader::Opening plugin:", file)
-		p, err := plugin.Open(file)
+	var pluginFiles []string
+	if pluginFilesPath != "" {
+		for _, file := range strings.Split(pluginFilesPath, ":") {
+			if strings.HasSuffix(file, ".so") {
+				pluginFiles = append(pluginFiles, filepath.Join(pluginDirPath, file))
+			}
+		}
+	} else {
+		files, err := os.ReadDir(pluginDirPath)
 		if err != nil {
-			println("custom_loader::Error opening plugin:", err.Error())
+			log.Err("custom_loader::Error reading plugin directory:", err)
 			return
 		}
-		println("custom_loader::Plugin opened successfully")
+		for _, file := range files {
+			if strings.HasSuffix(file.Name(), ".so") {
+				pluginFiles = append(pluginFiles, filepath.Join(pluginDirPath, file.Name()))
+			}
+		}
+	}
 
-		initFunc, err := p.Lookup("InitPlugin")
+	for _, file := range pluginFiles {
+		log.Info("custom_loader::Opening plugin:", file)
+		p, err := plugin.Open(file)
 		if err != nil {
-			println("custom_loader::Error looking up init function:", err.Error())
-			return
+			fmt.Println("custom_loader::Error opening plugin:", err.Error())
 		}
+		fmt.Println("custom_loader::Plugin opened successfully")
 
-		if initFunc, ok := initFunc.(func()); ok {
-			initFunc()
-			println("custom_loader::Plugin init function invoked successfully")
-		} else {
-			println("custom_loader::Symbol has wrong type")
+		getExternalComponentFunc, err := p.Lookup("GetExternalComponent")
+		if err != nil {
+			log.Info("custom_loader::Error looking up GetExternalComponent function:", err.Error())
+			os.Exit(1)
```

Comment:
then i think it should be consistent, because there are other components as well so either don't exit on any of custom component failure or fail on all. failing them on all is making more sense.